app/src/main/java/org/gdg/frisbee/android/chapter/MainActivity.java

```diff
@@ -130,6 +139,13 @@ public void onCreate(Bundle savedInstanceState) {
             }
 
             selectedChapterId = getChapterIdFromIntent(intent);
+
+            if (AppInviteReferral.hasReferral(intent)) {
+                // In this case the referral data is in the intent launching the MainActivity,
```

Comment:
I am keeping these currently since they are helpful. The integration is not fully tested and finished yet on my side. I will remove these and clean them up before merge. Sorry for the mess. :)